custom/abt/reports/late_pmt.py

```diff
@@ -139,66 +80,85 @@ def headers(self):
         )
 
     @cached_property
-    def smss_received(self):
-        data = SMS.objects.filter(
-            domain=self.domain,
-            couch_recipient_doc_type='CommCareUser',
-            direction=INCOMING,
-            couch_recipient__in=self.get_user_ids,
-            date__range=(
-                self.startdate,
-                self.enddate
-            )
-        ).exclude(
-            text="123"
-        ).values('date', 'couch_recipient').annotate(
-            number_of_sms=Count('couch_recipient')
-        )
-        return {(sms['date'].date(), sms['couch_recipient']) for sms in data}
-
-    @cached_property
-    def get_user_ids(self):
-        return [user['user_id'] for user in self.get_users]
-
-    @cached_property
-    def get_users(self):
-        return LatePMTUsers(config=self.report_config).get_data()
+    def pmts_submitted(self):
```

Comment:
Could this be done in chunks?  Alternatively, if `case.operation_site` is available in the form, that could make this much easier.

Reply:
Good call. eba2efae9a3